lte/gateway/c/session_manager/UEShard.cpp

```diff
@@ -0,0 +1,78 @@
+/**
+ * Copyright 2020 The Magma Authors.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <string>
+#include <utility>
+#include <vector>
+#include <algorithm>
+#include <map>
+
+#include "UEShard.h"
+#define MAX_SHARD_SIZE 100
+
+namespace magma {
+
+UEShard::UEShard() {
+  number_of_shards = 0;
+}
+
+int UEShard::add_ue(const std::string& imsi) {
+  int shard_id;
+  if (shards.size() == 0) {
+    shard_id = 0;
+    shards.push_back({imsi});
+    number_of_shards++;
+  } else {
+    for (auto i = 0; i < number_of_shards; i++) {
+      if (total_ues_for_shard(i) < max_shard_size) {
+        shards[i].push_back(imsi);
+        return i;
+      }
+    }
+    shard_id = number_of_shards++;
+    shards.push_back({imsi});
+  }
+  return shard_id;
+}
```

Comment:
rather than keeping imsi list per shard, it would be better to keep the shard ID in session-state for each imsi. since the shard-id needs to be persistent. it would save another persistent map in sessionD.

Reply:
once we move the shard-id to session-state, UEShard needs to keep track of stard level info, like count of UEs in each shard.